nomad/structs/funcs.go

```diff
@@ -136,6 +136,14 @@ func AllocsFit(node *Node, allocs []*Allocation, netIdx *NetworkIndex) (bool, st
 		return false, "bandwidth exceeded", used, nil
 	}
 
+	// Check devices
+	if checkDevices {
+		accounter := NewDeviceAccounter(node)
+		if accounter.AddAllocs(allocs) {
+			return false, "device oversubscribed", used, nil
```

Comment:
should we return the id of the oversubscribed device in the returned string?

Reply:
its hard since it may be multiple devices. We can improve it if it turns out to be an issue